SCAutolib/models/gui.py

```diff
@@ -0,0 +1,296 @@
+                break
+
+            # More than one word matches, choose the first match
+            # Probably deterministic, but it should not be relied upon
+            else:
+                print('Too many to choose from') # TODO replace the print function
+                item = df.iloc[0]
+
+        if time() >= end_time:
+            raise Exception('Found no matching key in any screenshot.')
+
+        x = int(item['left'] + item['width']/2)
+        y = int(item['top'] + item['height']/2)
+
+        self.mouse.move(x, y)
+        self.mouse.click()
+        sleep(1)
+
+    def assert_text(self, key: str, timeout=0):
+        '''
+        Given key must be found in a screenshot before the timeout.
+
+        If the key is not found, exception is raised.
+        Zero timeout means that only one screenshot will be taken and evaluated.
+        '''
+
+        end_time = time() + timeout
+        first = True
+
+        while first or time() < end_time:
+            first = False
+            # Capture the screenshot
+            screenshot = self.screen.screenshot()
+            df = self._image_to_data(screenshot)
+            selection = df['text'] == key
+
+            # The key was found
+            if selection.sum() != 0:
+                return
+
+        raise Exception('The key was not found.')
+
+
+    def assert_no_text(self, key: str, timeout=0):
+        '''
+        If the given key is found in any screenshot before the timeout, exception is raised.
+
+        Zero timeout means that only one screenshot will be taken and evaluated.
+        '''
+
+        end_time = time() + timeout
+        first = True
+
+        while first or time() < end_time:
+            first = False
+            # Capture the screenshot
+            screenshot = self.screen.screenshot()
+            df = self._image_to_data(screenshot)
+            selection = df['text'] == key
+
+            # The key was found, but should not be
+            if selection.sum() != 0:
+                raise Exception('The key was found in the screenshot.')
+
+
+    def wait_still_screen(self, time_still=5, timeout=30):
+        '''
+        Wait until the screen content stops changing.
+
+        When nothing on screen has changed for time_still seconds, continue.
+        If the screen content is changing permanently,
+        fail with exception after timeout seconds.
+        '''
+
+        timeout_end = time() + timeout
+        time_still_end = time() + time_still
+        last_screenshot = None
+
+        while time() < timeout_end and time() < time_still_end:
+            # Capture the screenshot
+            screenshot = None
+            while screenshot is None and time() < timeout_end:
+                # If capturing fails, try again
+                screenshot = self.screen.screenshot()
+
+            # If this is the first loop, there is no last screenshot
+            if last_screenshot is None or not self._images_same(screenshot, last_screenshot):
```

Comment:
`last_screenshot` variable isn't change after initialistation on line 262, so check for None value probably useless